src/Egulias/EmailValidator/Parser/Parser.php

```diff
@@ -36,14 +36,18 @@ protected function validateQuotedPair()
     }
 
     /**
-     * @return string the the comment
-     * @throws \InvalidArgumentException
+     * @param int $openedParenthesis
+     * @return string the comment
      */
-    protected function parseComments()
+    protected function parseComments(&$openedParenthesis = 0)
```

Comment:
And making it protected within `Parser` class? We can initialize it to 0 outside the loop of the `Local` or `Domain` parsers.
Similar side-efectish but more OOP.
WDYT?